### PR TITLE
Salvage Expedition Rebalance (Follow up, p. 1)

### PR DESCRIPTION
**Pull request by ErhardSteinhauer**
### Spawners:
- Diluted filled container spawners (lockers, toolboxes, medkits, weapon cases and such) with empty ones.
- Reworked weapon spawners: now weapons are divided into 5 tiers of rarity. 
**T1** includes common guns (improvised and most of Liberation Station stuff) and melee weapons (mostly improvised) and upon initializing has 10% chance to spawn T2 instead, loose weapons in dungeons are T1;
**T2** introduces .45cal guns and some full-auto capable guns, melee weapons on T2 are dedicated melee weapons and upon initializing has 10% chance to spawn T3 instead; spawn in easier types of vaults in dungeons;
**T3** - rifles and advanced melee weapons, upon initializing has 10% chance to spawn T4 instead; spawn in harder types of vaults in dungeons;
**T4** energy swords, atreides smg, self recharging laser guns and other wonder tech, upon initializing has 1% chance to spawn T5 instead, not mapped;
**T5** - L6s China Lakes, RPGs and such, not mapped.
- Added armor (includes heavy armor, low chance though) and shield spawners  (includes telescopic shield, low chance though).
- Added hardsuit spawners: notably among other hardsuits SCAF and experimental salvager hardsuits now obtainable from expeditions.
- Added explosives to weapon loot pool: fire bombs, pipe bombs, explosive grenades (rare), seismic charges (extra rare), c4 (extra mega rare).
- Added more chems to chem spawners.

### Dungeon themes:
- Fixed Virology Lab dorms and "secret" chamber, replaced `AlwaysPowered` light sources.
- Slightly buffed loot on Snowy Labs dungeon theme.
- Buffed loot from Mineshaft and Haunted Caves dungeon themes in general. Mineshaft now has a better chances of providing salvager hardsuits, Haunted Caves have a better chances of providing improvised explosives.
- Added Salvage Outpost dungeon theme.

### Bundled changes:
- Fixed minor graphical error with long weapon case.

## To do
- [x] Fiddle with spawner percentages. Specifically with materials and weapons.
- [x] Add more weapons to spawners.
- [x] Add armor spawner.
- [x] More chems for chem spawner.
- [x] Fix secret chamber on virology lab dungeon theme.
- [x] Fix dorms on virology lab dungeon theme.
- [x] Look through dungeon themes again and fix troublesome rooms.
- [x] Add proper salvage outpost theme.

## Why / Balance
Slightly reducing entity spam, moderately increasing loot scarcity and loot quality.

## How to test
Dungeon diving! Or straight up use spawners.

## Media

- [x] This PR does not require an ingame showcase (but here are some salvage outpost dungeon theme screenshots).

## Breaking changes
none yet

**Changelog**
:cl: erhardsteinhauer (discord)
- add: Added Salvage Outpost dungeon theme.
- tweak: Changed expedition loot pools: items should become slightly more scarce but of better value overall.
- tweak: Added advanced hardsuits to expedition loot pool tables.
- tweak: Weapon cases now provide moderate explosion protection for contents.
- fix: Fixed minor graphical error on long weapon case.

### CODE REVIEW

File: Resources/Prototypes/_NF/Entities/Markers/Spawners/Random/dungeon_items_weapons.yml [hunk 1]
```diff
@@ -0,0 +1,418 @@
...
+  noSpawn: true
+  components:
+  - type: Sprite
+    layers:
+    - state: green
+      scale: 0.7, 0.7
+    - sprite: Objects/Weapons/Bombs/seismic.rsi
+      state: icon
+    - sprite: Effects/crayondecals.rsi
+      state: 4
```
Comment (whatston3) [inline Resources/Prototypes/_NF/Entities/Markers/Spawners/Random/dungeon_items_weapons.yml:412]: Tier 5, state should be 5.
```suggestion
      state: 5
```

File: Resources/Prototypes/_NF/Catalog/Fills/Items/weapon_cases_expedition.yml [hunk 2]
```diff
@@ -233,6 +375,33 @@
...
+    contents:
+    - id: WeaponLauncherChinaLake
+      amount: 1
+    - id: GrenadeFrag
+      amount: 2
+
+# LMG
+- type: entity
+  parent: WeaponCaseLong
+  id: WeaponCaseLongLauncherL6
```
Comment (whatston3) [inline Resources/Prototypes/_NF/Catalog/Fills/Items/weapon_cases_expedition.yml:394]: `LightMachineGunL6` instead of `LauncherL6`?  Remember to change other instances if you do change this.

**Review by whatston3 — CHANGES_REQUESTED** [review]
A few requests for changes, nothing huge.  The dungeon weapon loot tables can cascade up multiple tiers, right?  If so, that's pretty fun!

I like the feeling that the empty lockers give, as though whoever was running the place had to leave in a disordered rush.  Tried to playtest a few expeditions, not really the greatest at that.  Still, the new salvage outpost dungeon type looks cool, just need to get my head around the new tiles.

I'd like to take a look at the maps this uses for dungeon generation, but I don't really know where to start there.  Had a bit of trouble loading them in-game, I'll take another shot at it in a bit.

Overall, exciting stuff, empty lockers/medkits seem like a fitting change.

File: Resources/Prototypes/_NF/Entities/Markers/Spawners/Random/dungeon_items_mercenary.yml [hunk 3]
```diff
@@ -353,6 +454,7 @@
     - VendingMachineArcadia
     - VendingMachineSec
     - VendingMachineSecDrobe
+    - SpawnDungeonMachineFrame
```
[the remark this answers is not in the document]
Comment (ErhardSteinhauer) [reply Resources/Prototypes/_NF/Entities/Markers/Spawners/Random/dungeon_items_mercenary.yml:457]: Those don't exist as separate entities

[the remark this answers is not in the document]
Comment (whatston3) [reply Resources/Prototypes/_NF/Entities/Markers/Spawners/Random/dungeon_items_mercenary.yml:457]: Fair, thanks.

File: Resources/Prototypes/_NF/Procedural/dungeon_configs.yml [hunk 4]
```diff
@@ -433,3 +434,85 @@
...
+          prob: 0.05
+          orGroup: content
+        - id: DefibrillatorCabinetFilled
+          prob: 0.1
+          orGroup: content
+
+    - !type:BoundaryWallPostGen
+      tile: FloorSteel
+      wall: WallMining
+      cornerWall: WallMining
```
[the remark this answers is not in the document]
Comment (ErhardSteinhauer) [reply Resources/Prototypes/_NF/Procedural/dungeon_configs.yml:488]: From my observations (which isn't backed up by statistics though) players when faced with a choice between dismantling a wall or digging some more in hopes of finding a window usually go for digging.
Also, yes, this is intended to be something that players can use their hoarded explosives.

But I guess I should add deconstruction graphs, because both windows and walls have high damage thresholds .

[the remark this answers is not in the document]
Comment (whatston3) [reply Resources/Prototypes/_NF/Procedural/dungeon_configs.yml:488]: Hm.  Guess I really don't do expeditions that often then.

Should probably be left for another PR, at any rate.

File: Resources/Prototypes/_NF/Entities/Objects/Tools/weapon_cases.yml [hunk 5]
```diff
@@ -72,5 +73,6 @@
       - EnergySword
       - ItemToggleDisarmMalus
       - Blocking
+      - Explosive
```
[the remark this answers is not in the document]
Comment (ErhardSteinhauer) [reply Resources/Prototypes/_NF/Entities/Objects/Tools/weapon_cases.yml:76]: I made weapon cases as a visual aid for players: some weapons are exceedingly hard to spot even without environmental modifiers that decrease visibility. I don't think that cases are used by players to store weapons or ammo, they usually just loot weapons from cases while leaving the cases behind. There is no mechanical benefits in using weapon cases for storage purposes.
What I'm leading with that is maybe this issue isn't a big deal and can be overlooked? Idk honestly.

File: Resources/Prototypes/_NF/Catalog/Fills/Items/weapon_cases_expedition.yml [hunk 6]
```diff
@@ -80,11 +190,13 @@
       amount: 1
     - id: MagazinePistolSubMachineGun
       amount: 2
+    - id: MagazineBoxPistol
+      amount: 2
```
[the remark this answers is not in the document]
Comment (ErhardSteinhauer) [reply Resources/Prototypes/_NF/Catalog/Fills/Items/weapon_cases_expedition.yml:194]: That and also because as of right now non-nfsd players don't have a sure way of procuring pistol ammo (ammo fabs have a chance to spawn, so not guaranteed)

File: Resources/Prototypes/_NF/Entities/Objects/Tools/weapon_cases.yml (hunk 5 above)
[the remark this answers is not in the document]
Comment (dvir001) [reply Resources/Prototypes/_NF/Entities/Objects/Tools/weapon_cases.yml:76]: FYI, This fails since fills happen on server side but the fill into storage is starting client side to server side I think.
So the explosives cannot be inserted based on comp acting server side and not shared

TLDR, While not a big error its a bit stinky.

[the remark this answers is not in the document]
Comment (ErhardSteinhauer) [reply Resources/Prototypes/_NF/Entities/Objects/Tools/weapon_cases.yml]: A'ight, I've added empty comp. No error messages upon inserting explosives in weapon cases. Also gave some explosion resistance to cases.

File: Resources/Prototypes/_NF/Catalog/Fills/Items/weapon_cases_expedition.yml (hunk 2 above)
Comment (ErhardSteinhauer) [reply Resources/Prototypes/_NF/Catalog/Fills/Items/weapon_cases_expedition.yml]: Done

File: Resources/Prototypes/_NF/Catalog/Fills/Items/weapon_cases_expedition.yml (hunk 6 above)
[the remark this answers is not in the document]
Comment (whatston3) [reply Resources/Prototypes/_NF/Catalog/Fills/Items/weapon_cases_expedition.yml:194]: Gotcha, thanks!

**Comment by ErhardSteinhauer** [comment]
> A few requests for changes, nothing huge. The dungeon weapon loot tables can cascade up multiple tiers, right? If so, that's pretty fun!

Yeah, that was the idea. I doesn't happen too often, but in my testing I encountered at least once RPG spawned outside vault, so the weapon spawner went from T1 all the way to T5.

> I like the feeling that the empty lockers give, as though whoever was running the place had to leave in a disordered rush. Tried to playtest a few expeditions, not really the greatest at that. Still, the new salvage outpost dungeon type looks cool, just need to get my head around the new tiles.

Yeah, those kinda turned out a bit different than I imagined: somewhat harder on the eyes tbh. But I'll wait for players feedback before changing walls and windows to something else I guess.

> I'd like to take a look at the maps this uses for dungeon generation, but I don't really know where to start there. Had a bit of trouble loading them in-game, I'll take another shot at it in a bit.

Dungeon "maps" are located in `/Maps/_NF/Dungeon/`, maps from base game aren't used. Here is example for loading a map: `mapping 111 /Maps/_NF/Dungeon/virology_lab.yml `

> Overall, exciting stuff, empty lockers/medkits seem like a fitting change. (Edit: bit of a nerf, but that seems like the idea here)

I would actually love to increase the weight of empty lockers even more, but with current system it looks clunky, hoping for this PR to be merged eventually: https://github.com/space-wizards/space-station-14/pull/28557

**Review by whatston3 — APPROVED** [review]
Changes look good, and the grenade/explosive weapon case insertion works as expected.  Nice!

**Review by dvir001 — APPROVED** [review]
